File: 2019-2020/pset6/more/mario.py
# ------------------------------------------------------------------------------
# mario.py
# "More comfortable" version
#
# Computer Science 50
# Problem Set 6
#
# Prints out a double half-pyramid of a specified height
#
# 2020:
# https://cs50.harvard.edu/x/2020/psets/6/mario/more/
# 2019
# https://docs.cs50.net/2019/x/psets/6/sentimental/mario/more/mario.html
# ------------------------------------------------------------------------------

from cs50 import get_int

while True:
    height = get_int("Height: ")
    if 0 < height < 9:
        break

for k in range(height):
    print('{0}{1}  {1}'.format((height - k - 1) * ' ', (k + 1) * '#'))

File: CS50_Introduction_to_Computer_Science/pset1/mario.c
/**
 * mario.c
 *
 * CS50 Problem Set 1: Mario
 *
 * Prints out a pyramid of a specified height
 */

#include <cs50.h>
#include <stdio.h>

const char SHARP = '#';
const char BLANC = ' ';
const int MIN_HEIGHT = 1;
const int MAX_HEIGHT = 8;

void chart(char c, int n)
{
    for (int i = 0; i < n; ++i)
    {
        putchar(c);
    }
}

int main(void)
{
    int n;

    do
    {
        n = get_int("Height: ");
    }
    while ((n < MIN_HEIGHT) || (n > MAX_HEIGHT));

    for (int i = 1; i <= n; ++i)
    {
        chart(BLANC, n - i);
        chart(SHARP, i);
        chart(BLANC, 2);
        chart(SHARP, i);
        printf("\n");
    }
    return 0;
}
